python/tests/simplify.py

```diff
@@ -582,12 +582,15 @@ def __init__(self, ts, sample, ancestors):
         for sample_id in sample:
             self.add_ancestry(0, self.sequence_length, sample_id, sample_id)
         self.edge_buffer = {}
+        self.oldest_ancestor = max(max(ancestors), max(sample))
 
     def link_ancestors(self):
         if self.ts.num_edges > 0:
             all_edges = list(self.ts.edges())
             edges = all_edges[:1]
             for e in all_edges[1:]:
+                if e.parent > self.oldest_ancestor:
```

Comment:
I don't think works in general, you're assuming that nodes have nondecreasing time. Better to check ``time[e.parent] > self.oldest_ancestor_time``.

Reply:
just checking, there isn't a max() function in any of the C libraries tskit relies on, right? if not I'll write that myself

Reply:
There's python's native `max` or `numpy.ndarray.max`?

Reply:
There's a TSK_MAX macro you can use in a loop.